doc/developer_guide/developer_guide.rst

```diff
@@ -0,0 +1,43 @@
+***************
+Developer Guide
+***************
+
+In this developer guide we explain how you can build this project, build the documentation and run the tests.
+
+#################################################
+Installation and Build/Setup (move/copy to user?)
+#################################################
+ - uses poetry
+
+#############
+Documentation
+#############
+
+This repository also uses itself to generate the documentation for the Generator. The Source Files for the documentation
+can be found in the `doc folder`_. Here is the :ref:`generated documentation <docu_start>` (You are already here ;) ).
+To build the documentation manually for testing, you can use:
+
+.. code:: bash
+
+    sphinx-apidoc -T -e -o api ../exasol_sphinx_github_pages_generator
+    sphinx-build -b html -W PathToProject/sphinx-github-pages-generator/doc .build
+
+#####
+Tests
+#####
+
+Tests are located in the `tests folder`_. Run them with
+
+    poetry run pytest tests
+
+The tests use a test-repository with a Machine user which is private in order not to confuse users. In order to run them,
```

Comment:
here you also mention the Test-Repo. I think you can remove the section for the Test-Repo in the design document